### DISCUSSION TITLE
Is it possible to create a custom DocCardList that displays items from different sidebars?

### DISCUSSION THREAD

**Discussion by Devon-White** [facebook/docusaurus#10752] [Q&A] [answered]
# Custom DocCardList Component for Cross-Sidebar Category Display

## Description
I'm trying to create a custom version of the `DocCardList` component that can display items from a different sidebar category than the current page. This would be useful for cross-referencing between different documentation sections.

## Current Behavior
The `DocCardList` component has two main ways of operation:

1. By default, it uses `useCurrentSidebarCategory()` to automatically display items from the current page's category
2. Alternatively, it accepts manual item definitions through the `items` prop:

```typescript
<DocCardList items={[
  {
    type: 'link',
    href: '/api/getting-started',
    label: 'Getting Started',
    description: '...'
  }
]} />
```

## Desired Behavior
Given a file structure like:

```
docs/
├── main-docs/          # Main documentation (uses mainSidebar)
│   ├── getting-started/
│   │   ├── installation.md
│   │   ├── quick-start.md
│   │   └── configuration.md
│   └── guides/
│       ├── basic-usage.md
│       └── advanced-features.md
└── api-docs/           # API documentation (uses apiSidebar)
    ├── getting-started/
    │   ├── api-overview.md
    │   └── authentication.md
    └── endpoints/
        ├── users.md
        └── products.md
```

I want to be able to reference items from a different sidebar category using a simple prop, like:

```typescript
<CustomDocCardList 
  docId="api-docs/endpoints/index" 
/>
```

This would allow me to display the endpoints category items while viewing a page in the main documentation, without manually maintaining the list of items.

## Questions
1. What would be the best approach to implement this functionality?
2. Are there any existing hooks or utilities that could help fetch sidebar data from a different category?

## Additional Context
- This would be particularly useful for cross-referencing between different documentation sections
- Ideally, the solution would automatically update when new documents are added to the referenced category

**Comment by slorber** [comment, marked as answer]
You can use `useDocsSidebar()` to get access to the sidedar in which the current doc is.

If you need to access other sidebars of a given version (not just categories), the only way I can think of is using `useDocsVersion().docsSidebars["apiSidebar"]`

Once you get the sidebar, it's your responsibility to traverse it and find the `category.items` to pass as props to `DocCardList`.

Remember all these are undocumented APIs on which we may make breaking changes. The API may break however I don't think the feature will disappear so it's relatively safe to use.

**Reply by Devon-White** [reply]
Appreciate the response!
